pkg/workflows/workflow.go

```diff
@@ -249,11 +252,19 @@ func (w *workflowLifecycle) submit(ctx context.Context, wp *unstructured.Unstruc
 		if err := controllerutil.SetControllerReference(w.addon, wfv1, w.scheme); err != nil {
 			return addonmgrv1alpha1.Failed, err
 		}
-
-		err = w.Create(ctx, wfv1)
+		wf, err := common.WorkFlowFromUnstructured(wfv1)
 		if err != nil {
+			fmt.Printf("### failed convert unstructure to workflow")
```

Comment:
No print statements, the needs to be an error log

Reply:
Yes, those old code needs a fix.

Reply:
28ea8a9de38232ea59393796dbe97d08d4dd8c6e